### DISCUSSION TITLE
Error while updating

### DISCUSSION THREAD

**Discussion by MikleBas** [magicbug/Cloudlog#1788] [General]
I saw another error in the logs during automatic updating

Commit or stash your changes before merging branches.
error: The specified untracked files in the working directory will be overwritten by the merge:
         assets/json/wwff.txt
Move these files or delete them before switching branches.
interrupt
Fast-forward failed; stopping here.

the specified file is empty **assets/json/wwff.txt**

**Comment by MikleBas** [comment]
Using /www/wwwroot/hamlog as working directory
Fetching changes from remote...
Fetched latest changes successfully
Fast-forwarding to HEAD (bleeding-edge)...
подсказка: Pulling without specifying how to reconcile divergent branches is
подсказка: discouraged. You can squelch this message by running one of the following
подсказка: commands sometime before your next pull:
подсказка:
подсказка:   git config pull.rebase false  # merge (the default strategy)
подсказка:   git config pull.rebase true   # rebase
подсказка:   git config pull.ff only       # fast-forward only
подсказка:
подсказка: You can replace "git config" with "git config --global" to set a default
подсказка: preference for all repositories. You can also pass --rebase, --no-rebase,
подсказка: or --ff-only on the command line to override the configured default per
подсказка: invocation.
Обновление 5bf89753..a048f085
error: Ваши локальные изменения в указанных файлах будут перезаписаны при слиянии:
        application/config/cloudlog.php
        application/config/migration.php
        application/config/mimes.php
        application/controllers/Dashboard.php
        application/controllers/Gridsquares.php
        application/controllers/Lotw.php
        application/controllers/Search.php
        application/controllers/Timeline.php
        application/controllers/User.php
        application/controllers/Visitor.php
        application/language/bulgarian/general_words_lang.php
        application/language/bulgarian/qslcard_lang.php
        application/language/chinese_simplified/general_words_lang.php
        application/language/chinese_simplified/qslcard_lang.php
        application/language/czech/general_words_lang.php
        application/language/dutch/general_words_lang.php
        application/language/english/general_words_lang.php
        application/language/english/qslcard_lang.php
        application/language/finnish/general_words_lang.php
        application/language/finnish/qslcard_lang.php
        application/language/french/general_words_lang.php
        application/language/french/qslcard_lang.php
        application/language/german/general_words_lang.php
        application/language/german/qslcard_lang.php
        application/language/german/qso_lang.php
        application/language/polish/general_words_lang.php
        application/language/polish/qslcard_lang.php
        application/language/spanish/general_words_lang.php
        application/language/spanish/qslcard_lang.php
        application/language/swedish/general_words_lang.php
        application/language/swedish/qslcard_lang.php
        application/language/turkish/general_words_lang.php
        application/language/turkish/qslcard_lang.php
        application/models/Csv_model.php
        application/models/Gridsquares_model.php
        application/models/Timeline_model.php
        application/models/User_model.php
        application/views/awards/cq/index.php
        application/views/awards/wwff/index.php
        application/views/contesting/index.php
        application/views/csv/data/export.php
        application/views/dashboard/index.php
        application/views/interface_assets/footer.php
        application/views/qso/edit_ajax.php
        application/views/search/filter.php
        application/views/search/main.php
        application/views/timeline/index.php
        application/views/user/add.php
        application/views/user/edit.php
        application/views/view_log/qso.php
        application/views/visitor/index.php
        assets/css/general.css
        assets/js/sections/contesting.js
        assets/js/sections/cqmap.js
        assets/js/sections/qso.js
        assets/json/satellite_data.json
Сделайте коммит или спрячьте ваши изменения перед слиянием веток.
error: Указанные неотслеживаемые файлы в рабочем каталоге будут перезаписаны при слиянии:
        assets/json/wwff.txt
Переместите эти файлы или удалите их перед переключением веток.
Прерываю
Fast-forward failed; stopping here.

**Comment by phl0** [comment]
> I saw another error in the logs during automatic updating
> 
> Commit or stash your changes before merging branches. error: The specified untracked files in the working directory will be overwritten by the merge: assets/json/wwff.txt Move these files or delete them before switching branches. interrupt Fast-forward failed; stopping here.
> 
> the specified file is empty **assets/json/wwff.txt**

That is because wwff.txt was added to the repo with 3de43250579fdca25612e379043368f666c3e116 in order to not fail if the file is empty. But in this case yours was updated on the older version and now contains local changes that should be overwritten. 

To fix you could do a `git checkout assets/json/wwff.txt`.

**Reply by MikleBas** [reply]
![error-git](https://user-images.githubusercontent.com/99086477/201956865-deddb4a6-079a-445d-89cf-76c68490946d.png)

**Reply by phl0** [reply]
> ![error-git](https://user-images.githubusercontent.com/99086477/201956865-deddb4a6-079a-445d-89cf-76c68490946d.png)

Are you in the root directory of the Cloudlog installation?

**Reply by MikleBas** [reply]
Yes

**Reply by phl0** [reply]
Do an `ls -al assets/json/` from there.

**Reply by MikleBas** [reply]
![5](https://user-images.githubusercontent.com/99086477/201967407-17a9ff7b-5c39-4254-b65a-879272664139.png)

**Reply by phl0** [reply]
Oh well. The file is 0 bytes?! But anyway `git checkout` should restore the original state of the file.

**Reply by MikleBas** [reply]
1. Executed your command, that file is still empty.
2. git gave advice - git pull
and text
Your branch is 89 commits behind "origin/master" and can be fast-forwarded.
   (use "git pull" to update your local branch)

**Reply by MikleBas** [reply]
I don't know how Git works at all - that's why I'm afraid to run the git pull command because of losing all the data (links are especially important)

**Reply by AndreasK79** [reply]
Well, if I read this correctly, it's only wwff.txt that is the problem. I would just delete wwff.txt, and then run git pull.

**Reply by MikleBas** [reply]
> Well, if I read this correctly, it's only wwff.txt that is the problem. I would just delete wwff.txt, and then run git pull.

Writes the following:
Commit or stash your changes before merging branches.
interrupt

**Reply by phl0** [reply]
What does `git status` give?

**Reply by MikleBas** [reply]
[git_autor.txt](https://github.com/magicbug/Cloudlog/files/10021810/git_autor.txt)

**Reply by phl0** [reply]
All those files were changed locally ... Did you copy a zip of the version 2 over the old master branch?

**Reply by MikleBas** [reply]
I didn’t copy anything myself, all that is there was an update script that ran on schedule from cron

**Reply by phl0** [reply]
Really strange ... maybe a failed git merge...
Try a `git stash`. That will revert all changes (assuming that there are no intended local changes).

**Reply by MikleBas** [reply]
Did
Working directory and index state saved WIP on master: 5bf89753 Revert "Merge pull request #1748 from florianprischl/patch-1"
what to do next?

**Reply by phl0** [reply]
It seems that you are on the master branch. Now if `git status` does not show any local changes any more you could do a `git pull` to update to the latest updates on master branch.

**Reply by MikleBas** [reply]
Yes, I did that already, now on the stable branch
True for some reason in the developer mode :)

**Reply by phl0** [reply]
> for some reason in the developer mode :)

See: https://github.com/magicbug/Cloudlog/discussions/1733

**Reply by MikleBas** [reply]
Well, the system was returned to the current state, but this still did not solve the problem with the certificate.
It's even embarrassing to distract you with your problems ..

**Reply by phl0** [reply]
No worries. But that is not Cloudlog related. The PHP code and command line curl cannot connect to clublog.org. I am not sure if Cloudlog can solve this. I even cannot reproduce this error on my system...
